ci/nodepool/scripts/prepare_node.sh

```diff
@@ -49,22 +51,35 @@ git clone https://github.com/pulp/pulp_packaging.git
 
 echo "Installing required puppet modules"
 sudo puppet module install --verbose --force puppetlabs-stdlib
-sudo puppet module install --verbose --force puppetlabs-mongodb
-sudo puppet module install --verbose --force ripienaar-module_data
-sudo puppet module install --verbose --force katello-qpid
 sudo puppet module install --verbose --force saz-sudo
 
+if [ ! "${VANILLA}" ] && [ ! "${DOCKER}" ]; then
+    sudo puppet module install --verbose --force puppetlabs-mongodb
+    sudo puppet module install --verbose --force ripienaar-module_data
+    sudo puppet module install --verbose --force katello-qpid
+
+    echo "Configuring pulp-unittest"
+    sudo puppet apply pulp_packaging/ci/deploy/utils/puppet/pulp-unittest.pp
+fi
+
 echo "Disable ttysudo requirement"
 sudo sed -i 's|Defaults[ ]*requiretty|#Defaults    requiretty|g' /etc/sudoers
 
 echo "Configuring jenkins_node_setup user"
 sudo puppet apply pulp_packaging/ci/deploy/utils/puppet/jenkins_node_setup.pp
 
-echo "Configuring pulp-unittest"
-sudo puppet apply pulp_packaging/ci/deploy/utils/puppet/pulp-unittest.pp
-
-if  [ "$OS_NAME" == "RedHatEnterpriseServer" ] && [ "$OS_VERSION" == "6" ]; then
-    echo "cleaning up rhel6"
+if [ "$OS_NAME" == "RedHatEnterpriseServer" ] && [ "$OS_VERSION" == "6" ] && [ ! "${DOCKER}" ]; then
     sudo rm -f /etc/udev/rules.d/70*
     sudo sed -i '/^\(HWADDR)\|UUID\)=/d' /etc/sysconfig/network-scripts/ifcfg-eth0
 fi
+
+if [ "${DOCKER}" ]; then
+    sudo yum install -y docker
+    sudo systemctl enable docker
```

Comment:
This will start docker on the next boot, but not right away. This is what you want?